### PR TITLE
WiFi-1997-LEDs-Incorrect-Service-State

### PR DESCRIPTION
**Pull request by NagConnectus**
This Patch fixes leds incorrect service state issue, and correct the
label names for all wifi6 APs

Signed-off-by: Nagendrababu <nagendrababu.bonkuri@connectus.ai>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c [hunk 1]
```diff
@@ -362,11 +363,18 @@ static void led_handler(int type,
...
-						set_led_config("heartbeat", conf->key, conf->value, led_string, led_section);
+						set_led_config("heartbeat", conf->key, conf->value, led_string, led_name_final);
 					}
 					else if(!strcmp(conf->key, "led_off")) {
-						set_led_config("none", conf->key, conf->value, led_string, led_section);
+						set_led_config("none", conf->key, conf->value, led_string, led_name_final);
+					}
+					else if(!strcmp(conf->key, "default_state")) {
+						if(!strcmp(led_section, "wifi2g"))
+							set_led_config("phy1tx", conf->key, conf->value, led_string, led_name_final);
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c:375]: I am not sure wifi2g is phy1tx for all APs.  Similar for wifi5g is phy0tx.

For example, my ECW5211 has board.json has phy0 for wifi2g.
root@OpenAp-b906af:/etc# cat board.json 
{
	"model": {
		"id": "edgecore,ecw5211",
		"name": "Edgecore ECW5211"
	},
	"led": {
		"wlan2g": {
			"name": "WLAN2G",
			"sysfs": "ecw5211:green:wlan2g",
			"trigger": "phy0tpt"
		},
		"wlan5g": {
			"name": "WLAN5G",
			"sysfs": "ecw5211:green:wlan5g",
			"trigger": "phy1tpt"
		}
	},
	"network": {
		"lan": {
			"ifname": "eth1",
			"protocol": "static"
		},
		"wan": {
			"ifname": "eth0",
			"protocol": "dhcp"
		}
	}
}

Also, what about EA8300 which has 3 radios and so likely has a phy2tx as well?

**Review by RickSommerville — COMMENTED** [review]
Mapping may vary between AP for default trigger.  See comment

**Review by RickSommerville — COMMENTED** [review]
See comment about mapping of events

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c (hunk 1 above)
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c:375]: Yes, I'll fix this issue based on the board and EA8300 doesn't have 2G and 5G LEDs.

Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c]: done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c [hunk 2]
```diff
@@ -335,6 +340,45 @@ static void set_led_config(char *trigger_name, char *key, char* value, char* led
...
+{
+	struct blob_attr *cur;
+
+	blob_buf_init(&b, 0);
+
+	if (!blobmsg_add_json_from_file(&b, DEFAULT_BOARD_JSON)) {
+		return NULL;
+	}
+
+	free(board_info);
```
[the remark this answers is not in the document]
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c:363]: done

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c [hunk 3]
```diff
@@ -353,7 +398,7 @@ static void led_handler(int type,
 	switch (type) {
 	case OVSDB_UPDATE_NEW:
 	case OVSDB_UPDATE_MODIFY:
-		if (!strcmp(conf->key, "led_blink") || !strcmp(conf->key, "led_off"))
+		if (!strcmp(conf->key, "led_blink") || !strcmp(conf->key, "led_off") || !strcmp(conf->key, "default_state"))
```
[the remark this answers is not in the document]
Comment (NagConnectus) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/command/src/node_config.c]: done

**Review by RickSommerville — APPROVED** [review]

**Comment by RickSommerville** [comment]
Merged 9796b90b, thanks.

**Pull request closed without merging**